### PR TITLE
Add FreeBSD support

### PR DESCRIPTION
**Pull request by JensHeinrich**
This patch adds support for freebsd using the `sysctl` command

### CODE REVIEW

File: autoload/battery/backend/freebsd.vim [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+" hw.acpi.battery.rate: present rate in mW
+" hw.acpi.battery.time: remaining time in minutes
+" hw.acpi.battery.life: percent capacity remaining
+
+let s:bat_status = 'hw.acpi.battery.state'
+let s:bat_capacity = 'hw.acpi.battery.life'
+
+function! s:read_sysctl(ctl) abort
+  let cmd = 'sysctl -n ' . a:ctl
+  return system(cmd)
```
Comment (lambdalisue) [inline autoload/battery/backend/freebsd.vim]: Please use `System.Job` instead to make it asynchronous. See other implementations or https://github.com/lambdalisue/vital-Whisky/blob/master/doc/Vital/System/Job.txt

**Review by lambdalisue — CHANGES_REQUESTED** [review]

File: autoload/battery/backend/freebsd.vim (hunk 1 above)
Comment (lambdalisue) [reply autoload/battery/backend/freebsd.vim]: Maybe https://github.com/lambdalisue/battery.vim/pull/13 would help

Comment (JensHeinrich) [reply autoload/battery/backend/freebsd.vim]: Done, though I had to realize `sysctl` output is hard to parse

File: autoload/battery/backend/freebsd.vim [hunk 2]
```diff
@@ -15,18 +15,33 @@
...
 
 function! s:freebsd_update() abort dict
-  let self.is_charging = s:read_sysctl(s:bat_status)
-  let self.value = s:read_sysctl(s:bat_capacity) + 0
+  if type(self.job) is# v:t_dict && self.job.status() ==# 'run'
+    return
+  endif
+  let data = []
+  let args = ['sysctl', '-n', s:bat_status, s:bat_capacity]
+  let self.job = s:Job.start(args, {
```
Comment (lambdalisue) [inline autoload/battery/backend/freebsd.vim]: Use `s:job` instead while users don't need to access the `job` instance through the `backend` instance.

File: autoload/battery/backend/freebsd.vim [hunk 3]
```diff
@@ -15,18 +15,33 @@
...
+
+function! s:on_exit(backend, buffer, exitval) abort
+  let content = join(a:buffer, '')
+  let a:backend.is_charging = content.getline(0) !=# 'NOT_CHARGING'
+  let a:backend.value = content.getline(1) + 0
 endfunction
 
 function! battery#backend#freebsd#define() abort
   return {
+   	\ 'job': 0,
```
Comment (lambdalisue) [inline autoload/battery/backend/freebsd.vim]: `v:null` is better for the default value for the `job`

```vim
let s:job = v:null
```

File: autoload/battery/backend/freebsd.vim [hunk 4]
```diff
@@ -15,18 +15,33 @@
...
+endfunction
+
+function! s:on_stdout(buffer, data) abort
+  call extend(a:buffer, a:data)
+endfunction
+
+function! s:on_exit(backend, buffer, exitval) abort
+  let content = join(a:buffer, '')
+  let a:backend.is_charging = content.getline(0) !=# 'NOT_CHARGING'
+  let a:backend.value = content.getline(1) + 0
```
Comment (lambdalisue) [inline autoload/battery/backend/freebsd.vim]: Do NOT use method syntax while Neovim 0.4 does not support it.

Rewrite above like:

```vim
let a:backend.is_charning = get(a:buffer, 0, '') !=# 'NOT_CHARNING'
let a:backend.value = get(a:buffer, 1, '') + 0
```

**Review by lambdalisue — CHANGES_REQUESTED** [review]

File: autoload/battery/backend/freebsd.vim (hunk 4 above)
Comment (JensHeinrich) [reply autoload/battery/backend/freebsd.vim]: Done, thanks for catching it

File: autoload/battery/backend/freebsd.vim [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+" hw.acpi.battery.rate: present rate in mW
+" hw.acpi.battery.time: remaining time in minutes
+" hw.acpi.battery.life: percent capacity remaining
+
+let s:bat_status = 'hw.acpi.battery.state'
+let s:bat_capacity = 'hw.acpi.battery.life'
+
+let s:Job = vital#battery#import('System.Job')
+
+let s:job = v:null
```
Comment (lambdalisue) [inline autoload/battery/backend/freebsd.vim]: It's minor but

1. Please use UPPER_CAMEL_CASE for constant variables
2. Reorder definitions to Vital modules, constant variables, script local variables

like

```suggestion
let s:Job = vital#battery#import('System.Job')

let s:BAT_STATUS = 'hw.acpi.battery.state'
let s:BAT_CAPACITY = 'hw.acpi.battery.life'

let s:job = v:null
```

**Review by lambdalisue — CHANGES_REQUESTED** [review]
One more minor fix please 🙏

File: autoload/battery/backend/freebsd.vim (hunk 5 above)
Comment (JensHeinrich) [reply autoload/battery/backend/freebsd.vim]: done

**Review by lambdalisue — APPROVED** [review]
Perfect 🎉

**Comment by lambdalisue** [comment]
Released as v1.1.0. Thanks again for your contribution 👍
